src/GraphQL/Server.hs

```diff
@@ -37,18 +38,16 @@ import GraphQL.Internal.Schema (HasName(..))
 import GraphQL.API
 import GraphQL.Internal.Input (CanonicalQuery)
 
-import Control.Monad.Catch (MonadThrow, throwM, Exception)
+import Control.Monad.Catch (Exception)
+
+data QueryError =
+  QueryError Text
+  | InternalError Text
```

Comment:
If necessary, we can leave a branch of the sum type to mean "once we do query validation properly, these should probably go away".